pkg/asset/agent/image/unconfigured_ignition.go

```diff
@@ -22,6 +23,34 @@ const (
 	unconfiguredIgnitionFilename = "unconfigured-agent.ign"
 )
 
+// GetConfigImageFiles returns the list of files or file paths to be included in the config-image.
+func GetConfigImageFiles() []string {
+	return []string{
+		"/etc/assisted/manifests/pull-secret.yaml", //nolint:gosec // not hardcoded credentials
+		"/etc/assisted/manifests/nmstateconfig.yaml",
+		"/etc/assisted/manifests/cluster-deployment.yaml",
+		"/etc/assisted/manifests/cluster-image-set.yaml",
+		"/etc/assisted/manifests/agent-cluster-install.yaml",
+		"/etc/assisted/extra-manifests", // all files in directory
+		"/etc/assisted/hostconfig",      // all files in directory
+		"/etc/assisted/hostnames",       // all files in directory
+		"/etc/issue",
+		"/etc/systemd/system.conf.d/10-default-env.conf",
+		"/root/.docker/config.json",
+		"/usr/local/bin/start-cluster-installation.sh",
```

Comment:
I think also start-cluster-installation.service should be included, as REQUIRED_MASTER_NODES/REQUIRED_WORKER_NODES env vars are now specified by the service.template: https://github.com/openshift/installer/commit/666dde5a975a2ed2c8d83c78abac2c9fc69d88be#diff-e185c956916cc81d4934f74f0d47973ba058cc5f52569ad91d4b9a131f38c84a.
I.e. otherwise, required_master_nodes/required_worker_nodes vars would be '0'.

Reply:
systemd services go in a different section of ignition from files.

Reply:
Yep. I had noted that needed to be done in https://redhat-internal.slack.com/archives/C02SPBZ4GPR/p1686658183214679?thread_ts=1686057590.009719&cid=C02SPBZ4GPR when https://github.com/openshift/installer/pull/7062 merged but forget to change it after it did. Will fix it

Reply:
Updated to copy the start-cluster-installation.service. The correct  node numbers are now being used. 
un 27 21:53:29 master-0.registry.ci.openshift.org start-cluster-installation.sh[3209]: Number of required master nodes: 3
Jun 27 21:53:29 master-0.registry.ci.openshift.org start-cluster-installation.sh[3209]: Number of required worker nodes: 0
Jun 27 21:53:29 master-0.registry.ci.openshift.org start-cluster-installation.sh[3209]: Total number of required nodes: 3

Note that this required using `systemctl daemon-reload` after the cpio in order for the updated env variables to be used.